tests/Jellyfin.Providers.Tests/Manager/ItemImageProviderTests.cs

```diff
@@ -0,0 +1,683 @@
+            BaseItem.FileSystem ??= Mock.Of<IFileSystem>();
+
+            var item = new MovieWithScreenshots();
+
+            var path = validPaths ? TestDataImagePath : "invalid path {0}";
+            for (int i = 0; i < count; i++)
+            {
+                item.SetImagePath(type, i, new FileSystemMetadata
+                {
+                    FullName = string.Format(CultureInfo.InvariantCulture, path, i),
+                });
+            }
+
+            return item;
+        }
+
+        private static ILocalImageProvider GetImageProvider(ImageType type, int count, bool validPaths)
+        {
+            var images = GetImages(type, count, validPaths);
+
+            var imageProvider = new Mock<ILocalImageProvider>();
+            imageProvider.Setup(ip => ip.GetImages(It.IsAny<BaseItem>(), It.IsAny<IDirectoryService>()))
+                .Returns(images);
+            return imageProvider.Object;
+        }
+
+        /// <summary>
+        /// Creates a list of <see cref="LocalImageInfo"/> references of the specified type and size, optionally pointing to files that exist.
+        /// </summary>
+        private static List<LocalImageInfo> GetImages(ImageType type, int count, bool validPaths)
+        {
+            var path = validPaths ? TestDataImagePath : "invalid path {0}";
+            var images = new List<LocalImageInfo>(count);
+            for (int i = 0; i < count; i++)
+            {
+                images.Add(new LocalImageInfo
+                {
+                    Type = type,
+                    FileInfo = new FileSystemMetadata
+                    {
+                        FullName = string.Format(CultureInfo.InvariantCulture, path, i)
+                    }
+                });
+            }
+
+            return images;
+        }
+
+        /// <summary>
+        /// Generates a <see cref="LibraryOptions"/> object that will allow for the requested number of images for the target type.
+        /// </summary>
+        private static LibraryOptions GetLibraryOptions(BaseItem item, ImageType type, int count)
+        {
+            return new LibraryOptions
+            {
+                TypeOptions = new[]
+                {
+                    new TypeOptions
+                    {
+                        Type = item.GetType().Name,
+                        ImageOptions = new[]
+                        {
+                            new ImageOption
+                            {
+                                Type = type,
+                                Limit = count,
+                                MinWidth = 0
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        // Create a class that implements IHasScreenshots for testing since no BaseItem class is also IHasScreenshots
+        private class MovieWithScreenshots : Movie, IHasScreenshots
+        {
+            // No contents
+        }
```

Comment:
This is now the only class in my workspace that is a `BaseItem` that implements `IHasScreenshots`.

Assuming it's not used in some other plugin that I don't have checked out, should either the handling for `IHasScreenshots` be removed or the interface itself be removed as unused?

Reply:
I wouldn't remove the interface, we never know if there is a plugin that implements it.

Reply:
I'm fine with removing it, AFAIK it was only used by the `Game` item type https://github.com/MediaBrowser/Emby.Common/blob/6ef8705ed609666555894c698dfc140fd8314831/MediaBrowser.Controller/Entities/Game.cs#L12